desktop/cmp/treemap/TreeMap.js

```diff
@@ -0,0 +1,297 @@
+highchartsExporting(Highcharts);
+highchartsOfflineExporting(Highcharts);
+highchartsExportData(Highcharts);
+highchartsTree(Highcharts);
+highchartsHeatmap(Highcharts);
+
+/**
+ * Component for rendering a TreeMap.
+ *
+ * It is a managed wrapper around a Highcharts TreeMap visualization, which renders
+ * records from a Store and optionally binds to a GridModel.
+ *
+ * @see TreeMapModel
+ */
+@HoistComponent
+@LayoutSupport
+export class TreeMap extends Component {
+
+    static propTypes = {
+        /** Primary component model instance. */
+        model: PT.oneOfType([PT.instanceOf(TreeMapModel), PT.object]).isRequired
+    };
+
+    static modelClass = TreeMapModel;
+
+    baseClassName = 'xh-treemap';
+
+    _chartElem = new Ref();
+    _chart = null;
+
+    constructor(props) {
+        super(props);
+
+        // Detect double-clicks vs single-clicks
+        this._clickCount = 0;
+        this._debouncedClickHandler = debounce(this.clickHandler, 500);
+
+        // Sync selection
+        this.addReaction({
+            track: () => [this.model.selectedIds, this.getMergedConfig()],
+            run: () => this.syncSelection(),
+            delay: 1 // Must wait for chart re-render on data / config change
+        });
+    }
+
+    render() {
+        // Default flex = 1 (flex: 1 1 0) if no dimensions / flex specified, i.e. do not consult child for dimensions;
+        const layoutProps = this.getLayoutProps();
+
+        if (layoutProps.width == null && layoutProps.height == null && layoutProps.flex == null) {
+            layoutProps.flex = 1;
+        }
+
+        this.renderHighChart();
+        this.updateLabelVisibilityAsync();
+
+        // Inner div required to be the ref for the chart element
+        return resizeSensor({
+            onResize: (e) => this.resizeChart(e),
+            item: box({
+                ...layoutProps,
+                className: this.getClassName(),
+                item: div({
+                    style: {margin: 'auto'},
+                    ref: this._chartElem.ref
+                })
+            })
+        });
+    }
+
+    //-------------------
+    // Implementation
+    //-------------------
+    renderHighChart() {
+        this.destroyHighChart();
+        const chartElem = this._chartElem.value;
+        if (chartElem) {
+            const config = this.getMergedConfig(),
+                parentEl = chartElem.parentElement;
+
+            assign(config.chart, {
+                width: parentEl.offsetWidth,
+                height: parentEl.offsetHeight
+            });
+
+            config.chart.renderTo = chartElem;
+            this._chart = Highcharts.chart(config);
+        }
+    }
+
+    resizeChart(e) {
+        const {width, height} = e[0].contentRect;
+        this._chart.setSize(width, height, false);
+        this.updateLabelVisibilityAsync();
+    }
+
+    destroy() {
+        this.destroyHighChart();
+    }
+
+    destroyHighChart() {
+        XH.safeDestroy(this._chart);
```

Comment:
This works because Highcharts has very nicely named their cleanup method `destroy()`, which is exactly what we expect / call in our XH method. Wondering though if we should be more explicit and call that method directly - we really intend the XH method to be called on other XH classes, and this might make it more clear that, yes, there really is a destroy method HC provides that we're calling appropriately.
```
if (this._chart) {
  this._chart.destroy();
  this._chart = null;
}
```